static-check/src/types/functions/function_input_variable.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright (C) 2019-2020 Aleo Systems Inc.
+// This file is part of the Leo library.
+
+// The Leo library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The Leo library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the Leo library. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::{Attribute, ParameterType, SymbolTable, Type, TypeError};
+use leo_typed::{FunctionInputVariable, Identifier, Span};
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
```

Comment:
as before, I'd advise against deriving `Hash` and `PartialEq` for the _whole_ struct if some parts of it are sufficient for checking for uniqueness and comparing to others